proxy/stat/server.py

```diff
@@ -1,12 +1,97 @@
+
+        self._holder_free_cnt_stat = StatGauge(
+            "op_resource_holder_free", "Operator holder accounts (free)", registry=registry
+        )
+        self._holder_used_cnt_stat = StatGauge(
+            "op_resource_holder_used", "Operator holder accounts (used)", registry=registry
+        )
+        self._holder_disabled_cnt_stat = StatGauge(
+            "op_resource_holder_disabled", "Operator holder accounts (disabled)", registry=registry
+        )
+        self._holder_blocked_cnt_stat = StatGauge(
+            "op_resource_holder_blocked", "Operator holder accounts (blocked)", registry=registry
+        )
+        self._holder_total_cnt_stat = StatGauge(
+            "op_resource_holder_total", "Operator holder accounts (total)", registry=registry
+        )
+
+        self._spending_tokens_balance: dict[SolPubKey, int] = {}
+        self._spending_tokens_balance_stat = StatGauge(
+            "op_resource_spending_tokens_balance", "Operator resource spending token balance", registry=registry
+        )
+
+    @AppDataApi.method(name="commitOpResourceEarnedTokensBalance")
+    def on_op_resource_earned_tokens_balance(self, data: OpResourceEarnedTokensBalanceData) -> None:
+        label = dict(token_name=data.token_name)
+
+        if data.token_name not in self._earned_tokens_balance:
+            self._earned_tokens_balance[data.token_name] = {}
+
+        self._earned_tokens_balance[data.token_name][data.eth_address] = data.balance
+        self._earned_tokens_balance_stat.set(label, sum(self._earned_tokens_balance[data.token_name].values()))
+
+    @AppDataApi.method(name="commitOpResourceHolderStatus")
+    def on_op_resource_holder_status(self, data: OpResourceHolderStatusData) -> None:
+        label = {}
+
+        self._holder_free_cnt[data.owner] = data.free_holder_cnt
+        self._holder_used_cnt[data.owner] = data.used_holder_cnt
+        self._holder_disabled_cnt[data.owner] = data.disabled_holder_cnt
+
+        self._holder_free_cnt_stat.set(label, sum(self._holder_free_cnt.values()))
+        self._holder_used_cnt_stat.set(label, sum(self._holder_used_cnt.values()))
+        self._holder_disabled_cnt_stat.set(label, sum(self._holder_disabled_cnt.values()))
+        self._holder_blocked_cnt_stat.set(label, data.blocked_holder_cnt)
+
+        self._holder_total_cnt_stat.set(
+            label,
+            sum(self._holder_free_cnt.values())
+            + sum(self._holder_used_cnt.values())
+            + sum(self._holder_disabled_cnt.values())
+            + data.blocked_holder_cnt,
+        )
+
+    @AppDataApi.method(name="commitOpResourceSpendingTokensBalance")
+    def on_op_resource_spending_tokens_balance(self, data: OpResourceSpendingTokensBalanceData) -> None:
+        label = {}
+        self._spending_tokens_balance[data.owner] = data.balance
+        self._spending_tokens_balance_stat.set(label, sum(self._spending_tokens_balance.values()))
```

Comment:
The same - the full picture:
- balance on each key
- total balance on all keys